### PR TITLE
Docs: specify that deathlink cause should contain the player name

### PR DESCRIPTION
**Pull request by alwaysintreble**
yes

### CODE REVIEW

**Comment by Jarno458** [comment]
This surprises me, this wasn't a requirement before, and i know several games that don't include it

Infact it used to be the opposite, not to include your player name in the case so you could do
```
					!string.IsNullOrEmpty(lastDeathLink.Cause)
						? $"DeathLink received from {lastDeathLink.Source}, Reason: {lastDeathLink.Cause}"
						: $"DeathLink received from {lastDeathLink.Source}");
```

**Comment by ScipioWright** [comment]
And I know several games that do include it, and so doing it the way you're doing it would show up strangely there.
Either the source needs to be included in the death link message or it needs to not be included. Including it makes it more flexible, such as being able to do "A grue thought Player looked tasty".

I guess there's the idea of, like in your example, specifically stating that you received a DeathLink. I don't really agree with the idea, though, since you need to explicitly enable death link in the first place, and that reduces creativity in death link messages.

**Comment by alwaysintreble** [comment]
when i implemented deathlink i asked how cause messages were supposed to be displayed, and what this change is doing is what i was told at the time and how clients are expected to handle it.

**Comment by ScipioWright** [comment]
Other people have been told different, probably because it wasn't documented anywhere.

**Review by Exempt-Medic — APPROVED** [review]
LGTM

**Review by black-sliver — COMMENTED** [review]
if you go all out with the spaces in the table, please also reformat the columns to properly align.

File: docs/network protocol.md [hunk 1]
```diff
@@ -674,8 +674,8 @@ Tags are represented as a list of strings, the common Client tags follow:
...
 
-| Name | Type | Notes |
-| ---- | ---- | ---- |
-| time | float | Unix Time Stamp of time of death. |
-| cause | str | Optional. Text to explain the cause of death, ex. "Berserker was run over by a train." |
-| source | str | Name of the player who first died. Can be a slot name, but can also be a name from within a multiplayer game. |
+| Name | Type | Notes                                                                                                                                                  |
+| ---- | ---- |--------------------------------------------------------------------------------------------------------------------------------------------------------|
+| time | float | Unix Time Stamp of time of death.                                                                                                                      |
+| cause | str | Optional. Text to explain the cause of death. When provided, or checked, this should contain the player name, ex. "Berserker was run over by a train." |
```
Comment (Jarno458) [inline docs/network protocol.md:680]: I would suggest removing the 'or checked' here as to me i have no idea what this is meant to mean

**Review by Jarno458 — CHANGES_REQUESTED** [review]

File: docs/network protocol.md (hunk 1 above)
Comment (alwaysintreble) [reply docs/network protocol.md:680]: when your client checks the death link packet to see what it should display.

File: docs/network protocol.md [hunk 2]
```diff
@@ -674,8 +674,8 @@ Tags are represented as a list of strings, the common Client tags follow:
 ### DeathLink
 A special kind of Bounce packet that can be supported by any AP game. It targets the tag "DeathLink" and carries the following data:
 
-| Name | Type | Notes |
-| ---- | ---- | ---- |
-| time | float | Unix Time Stamp of time of death. |
-| cause | str | Optional. Text to explain the cause of death, ex. "Berserker was run over by a train." |
-| source | str | Name of the player who first died. Can be a slot name, but can also be a name from within a multiplayer game. |
+| Name   | Type  | Notes                                                                                                                                                  |
+|--------|-------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (black-sliver) [inline docs/network protocol.md:678]:
```suggestion
| ------ | ----- | ------------------------------------------------------------------------------------------------------------------------------------------------------ |
```

Comment (black-sliver) [reply docs/network protocol.md:678]: Nvmd this suggestion. the other tables follow your style. I think we should keep consistency for now.

**Review by black-sliver — APPROVED** [review]

**Pull request merged**